drivers/platform/aducm3029/spi.c

```diff
@@ -0,0 +1,232 @@
+ *    notice, this list of conditions and the following disclaimer.
+ *  - Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ *  - Neither the name of Analog Devices, Inc. nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/************************* Include Files **************************************/
+/******************************************************************************/
+
+#include "spi_extra.h"
+#include "spi.h"
+#include "error.h"
+#include <stdlib.h>
+
+/******************************************************************************/
+/*****************************  Variables   **********************************/
+/******************************************************************************/
+
+static uint8_t	active_spi = 0; //Stores the active SPIs
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+
+/**
+ * @brief Allocates the memory needed for the descriptor
+ * @return A address to the allocated memory, NULL if the allocation fails.
+ */
+struct spi_desc *alloc_data_mem()
+{
+	struct spi_desc		*desc;
+	struct adicup_spi_desc	*extra;
+
+	desc = malloc(sizeof(struct spi_desc));
```

Comment:
calloc would be a better option since it initializes the memory to 0.

Reply:
Every malloc or only this one?

Reply:
You can have a look at latest platform specific drivers that are merged: https://github.com/analogdevicesinc/no-OS/blob/724c8dad5d1aedec46d1ce56c4b7fbeabe3504f2/drivers/platform/xilinx/spi.c#L59-L78

Generally, `calloc` is preferred over `malloc` since it is more risk-free.